x/ccv/parent/parent_test.go

```diff
@@ -501,3 +506,111 @@ func (s *ParentTestSuite) TestHandleConsumerDowntime() {
+	pKeep, cKeep := pApp.ParentKeeper, cApp.ChildKeeper
+	_ = pKeep
+
+	// Get the receiving fee pool on the provider chain
+	fcAddr := pApp.ParentKeeper.GetFeeCollectorAddressStr(pChain.GetContext())
+
+	// Ensure that the provider fee pool address stored on the consumer chain
+	// is the correct address
+	fcAddr2 := cApp.ChildKeeper.GetProviderFeePoolAddrStr(cChain.GetContext())
+	s.Require().Equal(fcAddr, fcAddr2)
+
+	// make sure we're starting at consumer height 16 (some blocks commited during setup)
+	s.Require().Equal(int64(16), cChain.GetContext().BlockHeight())
+
+	// get last consumer transmission
+	ltbh, err := cKeep.GetLastTransmissionBlockHeight(cChain.GetContext())
+	s.Require().NoError(err)
+	s.Require().Equal(int64(0), ltbh.Height)
+
+	bpdt := cKeep.GetBlocksPerDistributionTransmission(cChain.GetContext())
+	s.Require().Equal(int64(1000), bpdt)
+
+	// check the consumer chain fee pool
+	consumerFeePoolAddr := cApp.AccountKeeper.GetModuleAccount(
+		cChain.GetContext(), authtypes.FeeCollectorName).GetAddress()
+	tokens := cApp.BankKeeper.GetAllBalances(cChain.GetContext(), consumerFeePoolAddr)
+	s.Require().Len(tokens, 1)
+	s.Require().Equal(tokens[0].Denom, "stake")
+	s.Require().Equal(tokens[0].Amount, sdk.NewInt(205974705409))
+
+	//err = s.path.EndpointA.UpdateClient()
+	//err = s.path.EndpointB.UpdateClient()
+
+	ctx := cChain.GetContext()
+	sourcePort := transfertypes.PortID
+	sourceChannel := cKeep.GetDistributionTransmissionChannel(ctx)
+	sourceChannelEnd, found := cApp.IBCKeeper.ChannelKeeper.GetChannel(ctx, sourcePort, sourceChannel)
+	s.Require().True(found)
+	destinationChannel := sourceChannelEnd.GetCounterparty().GetChannelID()
+	fmt.Printf("debug destinationChannel: %v\n", destinationChannel)
+
+	// Commit some new blocks (commit blocks less than the distribution event blocks)
+	s.coordinator.CommitNBlocks(cChain, 1000-16)
+	err = s.path.EndpointB.UpdateClient()
+
+	// Commit some new blocks on the parent chain too
+	//s.coordinator.CommitNBlocks(pChain, 10)
+	//err = s.path.EndpointA.UpdateClient()
+
+	// check the consumer chain fee pool (should have increased
+	tokens = cApp.BankKeeper.GetAllBalances(cChain.GetContext(), consumerFeePoolAddr)
+	s.Require().Len(tokens, 1)
+	s.Require().Equal(tokens[0].Denom, "stake")
+	s.Require().Equal(tokens[0].Amount, sdk.NewInt(206083686059))
+
+	// check the provider chain fee pool
+
+	ctx = cChain.GetContext()
+	ltbh, err = cKeep.GetLastTransmissionBlockHeight(ctx)
+	s.Require().NoError(err)
+	bpdt = cKeep.GetBlocksPerDistributionTransmission(ctx)
+	curHeight := ctx.BlockHeight()
+	fmt.Printf(
+		"before distribution:\nltbh:%v\nbpdt:%v\ncurrHeight:%v\n(curHeight - ltbh.Height) < bpdt:%v\n",
+		ltbh.Height, bpdt, curHeight, (curHeight-ltbh.Height) < bpdt)
+
+	// Verify that the destinationChannel exists
+	// XXX if this doesn't exist then the transfer logic will fail when
+	// a the distribution transfer is invoked in the next block.
```

Comment:
@AdityaSripal The transfer `destinationChannel` doesn't exist at this point within my test (as you'll see a few lines down... This snipped from of lines 586-592 basically replicates the details of failing code that occurs when I actually try to submit a transfer (which is triggered by distribution once the next block is committed) - hence if I can figure out why the destinationChannel doesn't exist here then failing transfer may succeed (provided no new errors arise)... I'm not totally sure if I've setup the test properly, or if the error lies in how the connection is setup during `ConnAck`... Appreciate your insight!

Reply:
So the handshake could not have completed by this point, because the CCV channel setup will only do the INIT step of the transfer channel.

In the real world, a relayer will then have to complete the rest of the handshake. So you have to mock the completion of the transfer channel handshake in your test setup before you get to this line.

It is completely expected destination channel doesn't exist yet, because INIT will not populate destination channel. That gets created on TRY, and sent back to the sender on ACK.

Reply:
Great, thanks Adiyta, your fixes make total sense to me.. didn't understand the limitations of the test suite